File: chor-extraction/src/main/java/ast/cc/Selection.java
package ast.cc;

/**
 * Created by fmontesi on 03/04/17.
 */
public class Selection extends CCNode
{
    private final String sender;
    private final String receiver;
    private final String label;
    private final CCNode continuation;

    public Selection( String sender, String receiver, String label, CCNode continuation )
    {
        this.sender = sender;
        this.receiver = receiver;
        this.label = label;
    }

    public String sender()
    {
        return sender;
    }

    public String receiver()
    {
        return receiver;
    }

    public String label()
    {
        return label;
    }

    public <T> T accept( CCVisitor<T> visitor )
    {
        return visitor.visit( this );
    }
}

File: pe/src/main/kotlin/ast/cc/nodes/Selection.kt
package ast.cc.nodes

import ast.cc.interfaces.Interaction

/**
 * Created by fmontesi on 03/04/17.
 */
data class Selection(override val sender: String, override val receiver: String, val label: String) : Interaction {
    override fun toString(): String = "$sender->$receiver[$label]"
}
